### PR TITLE
Correctly rotate and flip image using Exif attributes.

### LINKED ISSUE #439: [BUG] - CropRect has mixed up width / height ⚠️ ⚠️
Using a Pixel 6 XL.

1. Click on sample of customactivity
2. Select `Camera` as the source
3. Take a portrait! picture
4. Have a breakpoint in `onSetImageUriComplete`: https://github.com/CanHub/Android-Image-Cropper/blob/85c98bbda8ec2024b9572518baaa894ac7ea55d5/cropper/src/main/java/com/canhub/cropper/CropImageActivity.kt#L274

Expected behavior: `view.cropRect` should have the correct dimensions (height > width)
Actual behavior: `view.cropRect` mixed up height with width (I get width of 3226 & height of 1814)

### PR DESCRIPTION
**Pull request by vanniktech**
Fixes #439

### CODE REVIEW

File: cropper/src/main/java/com/canhub/cropper/BitmapUtils.kt [hunk 1]
```diff
@@ -74,17 +74,12 @@ internal object BitmapUtils {
...
-        var ei: ExifInterface? = null
-        try {
-            val `is` = context.contentResolver.openInputStream(uri!!)
-            if (`is` != null) {
-                ei = ExifInterface(`is`)
-                `is`.close()
-            }
-        } catch (ignored: Exception) {
+    fun orientateBitmapByExif(bitmap: Bitmap?, context: Context, uri: Uri): RotateBitmapResult {
+        val exifInterface = context.contentResolver.openInputStream(uri)?.use { ExifInterface(it) }
```
Comment (vanniktech) [inline cropper/src/main/java/com/canhub/cropper/BitmapUtils.kt:78]: Now we close every single time just like in https://github.com/CanHub/Android-Image-Cropper/pull/440

File: cropper/src/main/java/com/canhub/cropper/BitmapUtils.kt [hunk 2]
```diff
@@ -106,7 +102,11 @@ internal object BitmapUtils {
             orientationAttributeInt == ExifInterface.ORIENTATION_TRANSPOSE
         val flipVertically = orientationAttributeInt == ExifInterface.ORIENTATION_FLIP_VERTICAL ||
             orientationAttributeInt == ExifInterface.ORIENTATION_TRANSVERSE
-        return RotateBitmapResult(bitmap, degrees, flipHorizontally, flipVertically)
+
+        return when (bitmap) {
+            null -> RotateBitmapResult(null, degrees, flipHorizontally, flipVertically)
+            else -> RotateBitmapResult(rotateAndFlipBitmapInt(bitmap, degrees, flipHorizontally, flipVertically), 0, false, false)
```
Comment (vanniktech) [inline cropper/src/main/java/com/canhub/cropper/BitmapUtils.kt:108]: calling `rotateAndFlipBitmapInt` is the fix and once we've flipped and rotated we no longer need to rotate / flip hence `0, false, false`

**Comment by vanniktech** [comment]
Hmm seems like cropping afterwards is broken. Will need to dig a bit further

**Comment by vanniktech** [comment]
I've tried but I gave up for now. The moment I change something, another feature breaks. It seems like everything is calibrated to behave on certain not so true data.

For the time being, I'll just copy what's inside `rotateAndFlipBitmapInt` and do that before passing a bitmap / uri to the library, then it works 🤷

**Comment by thekingrenz23** [comment]
This should already be good to go right?

**Comment by vanniktech** [comment]
Unfortuantely not. The cropping logic which crops the image after you've resized / rotated ect relies on the broken image and when fixing it and rotating it, like I've done here the output is just wrong. I haven't had time to look into this yet plus I've got my workaround which seems to be working.

**Comment by vanniktech** [comment]
I'll get back to it in the future.

**Pull request closed without merging**